subcommands/devices/delete_denied.go

```diff
@@ -0,0 +1,37 @@
+package devices
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func init() {
+	cmd.AddCommand(&cobra.Command{
+		Use:   "delete-denied <uuid> [<uuid>...]",
+		Short: "Remove a device UUID from the deny list",
+		Run:   doDeleteDenied,
+		Args:  cobra.MinimumNArgs(1),
+		Long: `Remove a device UUID from the deny list so that the UUID can be re-used.
+This is handy for Factories using HSMs and a factory-registration-reference
+server.`,
+	})
+}
+
+func doDeleteDenied(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	logrus.Debug("Deleting %r", args)
+
+	for _, uuid := range args {
+		fmt.Printf("Deleting %s .. ", uuid)
+		if err := api.DeviceDeleteDenied(factory, uuid); err != nil {
+			fmt.Printf("failed\n%s", err)
+			os.Exit(1)
+		} else {
+			fmt.Printf("ok\n")
+		}
```

Comment:
Why not `DieNotNil`?

Also, it is the first command which prints 'ok' in case of success. All other commands like this print nothing.

Reply:
i copied the code as-is from delete.go. i think it was done like this because it can remove multiple, so just dying isn't going to show you which device it failed on.

Reply:
Maybe, it is an instance of code I missed when refactoring to use DieNotNil everywhere; maybe it was added later; I don't know.
What I know is that `commands.DieNotNil(api.DeviceDelete(factory, name), "failed"); fmt.Println("ok")` is equivalent; but I'd prefer `commands.DieNotNil(api.DeviceDelete(factory, name))` as I see no value in these `failed, ok` things.

Just dying will show clearly on which device it failed due to these `Deleting xyz...` messages.